Library/Homebrew/test/support/helper/spec/integration_command.rb

```diff
@@ -0,0 +1,92 @@
+require "open3"
+
+RSpec::Matchers.define_negated_matcher :not_to_output, :output
+RSpec::Matchers.define_negated_matcher :be_a_failure, :be_a_success
+
+RSpec::Matchers.define :be_a_success do
+  match do |actual|
+    if actual.is_a?(Proc)
+      status = actual.call
+    else
+      status = actual
+    end
```

Comment:
I think this would be cleaner as a ternary but won't push too hard if you or @MikeMcQuaid have strong opinions to the contrary.